cmd/llama_receiver/llama_receiver.go

```diff
@@ -0,0 +1,347 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cybertec-postgresql/pgwatch/v3/api"
+	"github.com/destrex271/pgwatch3_rpc_server/sinks"
+	"github.com/jackc/pgx/v5"
+	"github.com/rifaideen/talkative"
+)
+
+const contextString = `
+You are an expert in analyzing PostgreSQL database metrics. I will provide you with a set of measurements for a specific database metric. Based on this data, I need you to derive actionable insights and recommendations. Please focus solely on your observations and analysis, without detailing any methods.
+
+The metrics are:  
+{DATA}  
+
+Please deliver a comprehensive report with your insights and suggestions based on the provided measurements.
+`
+
+type Batch map[string][]*api.MeasurementEnvelope
+
+type LlamaReceiver struct {
+	Client    *talkative.Client
+	Context   string
+	Ctx       context.Context
+	ServerURI string
+	DbConn    *pgx.Conn
+	AltDbConn *pgx.Conn
+	MsmtBatch Batch
+	BatchSize int
+	mu        sync.Mutex
+	sinks.SyncMetricHandler
+}
+
+type MeasurementsData struct {
+	DbID       uint
+	MetricName string
+	data       string
+}
+
+func NewLlamaReceiver(llmServerURI string, pgURI string, ctx context.Context, batchSize int) (recv *LlamaReceiver, err error) {
+	client, err := talkative.New(llmServerURI)
+	if err != nil {
+		log.Println("[ERROR]: unable to initialize llm client")
+		return nil, err
+	}
+
+	conn, err := pgx.Connect(ctx, pgURI)
+	if err != nil {
+		log.Println("[ERROR]: unable to connect to postgres instance")
+		return nil, err
+	}
+
+	// To use in insight generation to avoid any stuff
+	altConn, err := pgx.Connect(ctx, pgURI)
+	if err != nil {
+		log.Println("[ERROR]: unable to obtain additional connectionj")
+	}
+
+	recv = &LlamaReceiver{
+		Client:            client,
+		Context:           contextString,
+		Ctx:               ctx,
+		ServerURI:         llmServerURI,
+		DbConn:            conn,
+		AltDbConn:         altConn,
+		MsmtBatch:         make(Batch),
+		BatchSize:         batchSize,
+		SyncMetricHandler: sinks.NewSyncMetricHandler(1024),
+	}
+
+	err = recv.SetupTables()
+	if err != nil {
+		log.Println("[ERROR]: unable to setup tables: ", err)
+		return nil, err
+	}
+
+	return recv, nil
+}
+
+func (r *LlamaReceiver) SetupTables() error {
+	_, err := r.DbConn.Exec(r.Ctx, `CREATE TABLE IF NOT EXISTS Db(id serial Primary key, dbname varchar(255))`)
```

Comment:
```suggestion
	_, err := r.DbConn.Exec(r.Ctx, `CREATE TABLE IF NOT EXISTS Db(id serial Primary key, dbname text)`)
```
https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_varchar.28n.29_by_default

Reply:
damn, I was not aware of that 
Thanks!